### PR TITLE
Wealth distribution example

### PR DESCRIPTION
**Pull request by Datseris**
This Pr does many things:

1. Adds wealth distribution example, via the Literate syntax.
2. Changes the wealth distribution example from the Mesa one, to one that in my eyes makes a bit more sense, as the agents do not change place. `move_agent` is anyway showcased in the Schelling model.
3. Removes outdated files.
4. Uses UnicodePlot for one visualization (super cool)
5. Significantly improves `add_agent!`: it is now possible to do `add_agent!(model, properties...)` which adds a new agent to a random node.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -19,15 +19,21 @@ Agent.jl provides a structure and components for quickly implementing agent-base
...
+* __Schedulers__: users can choose from a range of activation regimes, i.e. the order with which agents activate, or implement a custom one.
 * __Spatial structures__: the framework implements 1D, 2D, and 3D grids which can optionally have periodic boundary conditions, meaning that edges of a grid connect to their opposite edges. An agent exiting from one edge enters the grid from the opposite edge. Moreover, users can construct irregular networks as the space where the agents live.
 * __Data collection__: users only specify the kind of data they need and the framework automatically collects them in a table. The collected data are then ready to be analyzed and visualized.
-* __Visualization__ users can create custom plots interactively from the simulation outputs using the [Data Voyager](https://github.com/vega/voyager) platform. Furthermore, they can visualize agent distributions on 2D grids.
 * __Batch run__: in agent-based modeling, we can rarely make conclusions from single simulation runs. Instead we run many replicates of a simulation and observe the mean behavior of the system. Agents.jl automates running simulation replicates and collecting and aggregating their results.
```
Comment (kavir1698) [inline docs/src/index.md]: We should also mention the `paramscan` feature.

**Comment by Datseris** [comment]
(@kavir1698 just so that we are clear: Does your 👍 mean that you _approve_ the PR, or you still need to go through a review?)

**Comment by kavir1698** [comment]
> Does your 👍 mean that you approve the PR, or you still need to go through a review?)

No, I am actually going through the review now.

**Comment by kavir1698** [comment]
Impressive work! Looks good to me (except mentioning `paramscan` in the features list). Thank you.

**Pull request merged**